File: onebot-client/src/main/java/cn/evole/onebot/sdk/response/misc/OcrResp.java
package cn.evole.onebot.sdk.response.misc;

import com.google.gson.annotations.SerializedName;
import lombok.Data;

import java.util.List;

/**
 * @author cnlimiter
 */
@Data
public class OcrResp {

    @SerializedName("texts")
    private List<TextDetection> texts;

    @SerializedName("language")
    private String language;

    @Data
    private static class TextDetection {

        /**
         * 文本
         */
        @SerializedName("text")
        private String text;

        /**
         * 置信度
         */
        @SerializedName("confidence")
        private int confidence;

        /**
         * 坐标
         */
        @SerializedName("coordinates")
        private int[][] coordinates;

    }

}

File: onebot/src/main/kotlin/sdk/response/misc/OcrResp.kt
package cn.evolvefield.onebot.sdk.response.misc

import com.google.gson.annotations.SerializedName

class OcrResp {
    @SerializedName("texts")
    var texts: List<TextDetection> = mutableListOf()

    @SerializedName("language")
    var language: String = ""

    class TextDetection {
        /**
         * 文本
         */
        @SerializedName("text")
        var text: String = ""

        /**
         * 置信度
         */
        @SerializedName("confidence")
        var confidence = 0

        /**
         * 坐标
         */
        @SerializedName("coordinates")
        var coordinates: Array<IntArray> = arrayOf()
    }
}
